test/jdk/sun/security/pkcs11/Mac/PBAMac.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2023, Red Hat, Inc.
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.math.BigInteger;
+import java.security.GeneralSecurityException;
+import java.security.Provider;
+import java.security.Security;
+
+import javax.crypto.Mac;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.interfaces.PBEKey;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.PBEParameterSpec;
+
+/*
+ * @test
+ * @bug 8301553
+ * @summary test password based authentication on SunPKCS11's Mac service
+ * @library /test/lib ..
+ * @run main/othervm/timeout=30 PBAMac
+ */
+
+public final class PBAMac extends PKCS11Test {
+    private static final char[] password = "123456".toCharArray();
+    private static final byte[] salt = "abcdefgh".getBytes();
```

Comment:
This converts the string into bytes using "default" charset which means this test will likely fail if the default charset changed unexpectedly? Same goes for other data such as 'plainText'.

Reply:
That's right. Fixed for salt and plainText.